src/components/table-view/components/TableViewRow.tsx

```diff
@@ -0,0 +1,169 @@
+/**
+ * TableViewRow Component
+ *
+ * Internal component for rendering a single table row with:
+ * - Selection checkbox
+ * - Cell content via renderers
+ * - Compound expansion support
+ * - Row actions
+ */
+
+import React from 'react';
+import { Tbody, Td, Tr } from '@patternfly/react-table/dist/dynamic/components/Table';
+import { ExpandableRowContent } from '@patternfly/react-table/dist/dynamic/components/Table';
+import type { CellRendererMap, ColumnConfigMap, ExpansionRendererMap } from '../types';
+
+export interface TableViewRowProps<TColumns extends readonly string[], TRow, TCompound extends string> {
+  /** Row data */
+  row: TRow;
+  /** Row index in the data array */
+  rowIndex: number;
+  /** Unique ID for this row */
+  rowId: string;
+  /** Column IDs */
+  columns: TColumns;
+
+  // Selection
+  /** Whether selection is enabled */
+  selectable: boolean;
+  /** Whether this row can be selected */
+  canSelect: boolean;
+  /** Whether this row is currently selected */
+  isSelected: boolean;
+  /** Callback when row selection changes */
+  onSelectRow?: (row: TRow, isSelecting: boolean) => void;
+
+  // Click
+  /** Whether this row is clickable */
+  isClickable: boolean;
+  /** Callback when row is clicked */
+  onRowClick?: (row: TRow) => void;
+
+  // Expansion
+  /** Currently expanded cell info */
+  expandedCell?: { rowId: string; column: TCompound } | null;
+  /** Expansion content renderers */
+  expansionRenderers?: ExpansionRendererMap<TCompound, TRow>;
+  /** Check if a cell can be expanded */
+  isCellExpandable: (row: TRow, column: TCompound) => boolean;
+  /** Callback when expansion is toggled */
+  onToggleExpand?: (row: TRow, column: TCompound) => void;
+
+  // Config
+  /** Set of compound column IDs */
+  compoundColumnSet: Set<string>;
+  /** Column configuration */
+  columnConfig: ColumnConfigMap<TColumns>;
+  /** Cell content renderers */
+  cellRenderers: CellRendererMap<TColumns, TRow>;
+  /** Render actions for this row */
+  renderActions?: (row: TRow) => React.ReactNode;
+  /** Total column count (for colspan) */
+  columnCount: number;
+}
+
+/**
+ * Internal row component for TableView.
+ * Renders a single row with cells, expansion, and actions.
+ */
+export function TableViewRow<TColumns extends readonly string[], TRow, TCompound extends string>({
```

Comment:
I might be overly cautious, but can we split this into a regular row and an expandable row? The expandable row would obviously use the regular as a internal component.